presto-main/src/main/java/com/facebook/presto/server/ServerInfoResource.java

```diff
@@ -92,4 +94,17 @@ public NodeState getServerState()
             return ACTIVE;
         }
     }
+
+    @GET
+    @Path("coordinator")
+    @Produces(TEXT_PLAIN)
+    public Response getServerCoordinator()
```

Comment:
Related, I just put up #5428 which makes the statement resource (`/v1/statement`) only available on the coordinator. This means if you do a `HEAD` to that URI, you'll get "405 Method Not Allowed" on the coordinator (it only allows `POST`) and a "404 Not Found" on the workers.